proof/invariant-abstract/AARCH64/ArchTcb_AI.thy

```diff
@@ -21,23 +21,19 @@ lemma activate_idle_invs[Tcb_AI_asms]:
   by (simp add: arch_activate_idle_thread_def)
 
 
-lemma empty_fail_getRegister [intro!, simp, Tcb_AI_asms]:
-  "empty_fail (getRegister r)"
-  by (simp add: getRegister_def)
-
+declare getRegister_empty_fail [Tcb_AI_asms]
```

Comment:
Does this mean that this is now proven before this file? If so, could it be removed from the locale completely, and then this declaration wouldn't be needed?

Reply:
It's only proved before this file in AARCH64 (in some crunch somewhere), but it's still in the interface, because other architectures have not necessarily seen that proof before at this point.

Reply:
Oh right, of course. Sorry, that should have been obvious.
I guess that this is a basic enough lemma that it would be nice to have this consistent on all the architectures and then out of the interface, but that would be pretty low priority.

Reply:
Hm, I'm not entirely sure about that. If we do explicitly prove it earlier in the other architectures, we should have some interface somewhere that documents that we want it to exist at a certain point, otherwise we might delete it again thinking it can be done as part of some later `crunch` only to then fail deep in some other seemingly unrelated proof which could be different per platform.

I guess all that is fixable with a comment, so maybe not so bad. But I'd still prefer fewer "works by carefully constructed accident on all architectures" lemmas.

Reply:
I definitely agree that it should be forced to exist, but the interface could be a more lightweight `requalify_facts` instead of being part of the `Tcb_AI` locale. This would be both less text and also require less duplicated boilerplate between all of the architectures.

The downside is that it wouldn't explicitly state what the lemma should look like. In this case `empty_fail` doesn't have preconditions so that shouldn't matter, but I can imagine other cases where lemmas could be accidentally formulated differently on different architectures.

Reply:
`requalify_facts` is a good idea, that would indeed fit everything we need at that point. I'll probably do that in a separate PR.